src/main/java/nom/tam/fits/HeaderCard.java

```diff
@@ -786,14 +787,18 @@ private HeaderCard(String key, String value, String comment, boolean nullable, b
             throw new HeaderCardException("Keyword too long");
         }
         if (value != null) {
-            value = value.replaceAll(" *$", "");
+            // Discard trailing spaces (TODO do we really want to?)
+            value = value.replaceAll("[ \t\r\n]*$", "");
 
+            /*
             if (value.startsWith("'")) {
                 if (value.charAt(value.length() - 1) != '\'') {
                     throw new HeaderCardException("Missing end quote in string value");
                 }
                 value = value.substring(1, value.length() - 1).trim();
```

Comment:
Just commenting out this code may be confusing to later readers. I would suggest either removing it altogether, or adding a comment about what's gone on here.

Reply:
Yes, that's necessary because it is used to parse quoted string from the header, i.e.:

```
 MYKEY   = 'some string value ' / Comment here
```
However, this method should not be called when constructing a card with a `HeaderCard()` constructor with a `String` value given programmatically. And that is the change I made (among other things..)

I also added test cases for quoted values in the string value itself. So I think we are good. 

However, I am still working on the parsing of header lines, because there is a lot of wonkiness there, and it's more trouble waiting to happen...

Reply:
I'm not objecting to the revised implementation, just the readability of the resulting source file.  Following the commit, lines 793-800 are just commented out, so will likely make it harder for future readers attempting to understand or maintain this class (why were these lines commented out? why were they there in the first place?)  So I am suggesting either removing those lines altogether (my preferred option) or adding some comment text nearby that explains why they're no longer required.

Reply:
Oh yes, sorry -- I misunderstood. Yes, I meant to remove the commented lines, just forgot... Thanks! :-D